tools/infer_manifest_test.go

```diff
@@ -0,0 +1,74 @@
+package tools
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"testing"
+)
+
+func TestInferManifestFile(t *testing.T) {
+	tests := []struct {
+		name         string
+		createsFile  bool
+		createsDir   bool
+		inputString  string
+		expectedPath string
+		expectErr    bool
+	}{
+		{"input, valid manifest in pwd", true, false, "manifest.yaml", "manifest.yaml", false},
+		{"input, invalid manifest in pwd", false, false, "substream.yaml", "", true},
+		{"input, valid dir with manifest", true, true, "manifests", "manifest/substreams.yaml", false},
+		{"input, invalid dir", false, false, "manifests", "", true},
+		{"input, valid dir w/o manifest", false, true, "manifests", "", true},
+		{"no input, has manifest", true, false, "", "substreams.yaml", false},
+		{"no input, no manifest", false, false, "", "", true},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			var file *os.File
+			var dir string
+			var err error
+
+			if test.createsDir {
+				dir, _ = os.MkdirTemp("", test.inputString)
```

Comment:
The `t *testing.T` has an helper `t.TempDir()` that you can use, it will be automatically cleaned up when done with it.

Reply:
So, you need to call `ResolveManifestFile` somewhere, it's the purpose of the test, so general flow is:

- Prepare test environment
- Call your function with test case input(s)
- Validate output so it matches the expected values
- Clean up if needed

I see no call to `ResolveManifestFile` here, so not sure what we are testing :)